### PR TITLE
feat(frozen-panel): 增加行列头冻结配置面板

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

**基于 https://github.com/antvis/S2/pull/2779 创建的分支, 所以含有 #2779 的代码**, 先看 #2779

✨ Feature

- [x] New feature

### 📝 Description

```tsx
  <FrozenPanel
    title="冻结行列头"
    defaultCollapsed={false}
    inputNumberProps={{
      size: 'small',
      step: 1,
    }}
    defaultOptions={{
      frozenRowHeader: true,
      frozenRow: [1, 2],
      frozenCol: [1, 2],
    }}
    onChange={(options) => {
      console.log('onChange:', options);
    }}
    onReset={(options, prevOptions) => {
      console.log('onReset:', options, prevOptions);
    }}
    />
```

### 🖼️ Screenshot

Kapture 2024-06-19 at 16 39 51

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-react-components/src/components/config/frozen-panel/frozen-input-number/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const PRE_CLASS = `${S2_PREFIX_CLS}-frozen-input-number`;
+
+export const FrozenInputNumber: React.FC<FrozenInputNumberProps> = React.memo(
+  (props) => {
+    const { disabled, value, onChange, ...attrs } = props;
+    const [inputValue, setInputValue] = React.useState<number | string | null>(
+      value,
+    );
+
+    const onDebounceChange = debounce((nextValue) => {
```
Comment (wjgogogo) [inline packages/s2-react-components/src/components/config/frozen-panel/frozen-input-number/index.tsx:16]: callback包一下？不然每次都是一个新的，不会有debounce效果

Comment (lijinke666) [reply packages/s2-react-components/src/components/config/frozen-panel/frozen-input-number/index.tsx]: 已修改

**Review by wjgogogo — APPROVED** [review]

**Pull request merged**